utilities/src/bits.rs

```diff
@@ -41,14 +41,14 @@ pub trait ToBits: Sized {
 
     /// Returns `self` as a boolean array in little-endian order.
     fn to_bits_le(&self) -> Vec<bool> {
-        let mut bits = vec![];
+        let mut bits = Vec::with_capacity(32);
```

Comment:
this value was chosen empirically; the same applies to the remaining pre-allocations